controllers/authrealm/suite_test.go

```diff
@@ -485,6 +488,142 @@ var _ = Describe("Process AuthRealm: ", func() {
 		})
 	})
 
+	It("process a AuthRealm CR with 3 identiity providers", func() {
```

Comment:
Would it make sense to consider (and is it possible to implement) a smaller test that just validates that OIDC is properly handled in the `createDexConnectors` function? I don't have much experience at all in this space, but it doesn't feel ideal to have to copy paste this huge test when really we're just validating the changes to that one function. Thoughts? What can we do here?

Reply:
sure.. I can just only validate OIDC connector. I added all 3 because we added new connector. But make sense we already have test for 2 connectors in authrealm. I'll modify tests

Reply:
I was thinking that rather than run the Reconcile function and test the outputs we could just run the `createDexConnectors` function directly and validate the output. If not now, perhaps we could consider this level of testing in the future. At a min let's update the `process a AuthRealm CR with 3 identiity providers` label.

Reply:
Yeah we can write these tests in the separate PR. I think we should write unit tests for all methods.